extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuarkusQuartzConnectionPoolProvider.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.quartz.runtime;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+
+import javax.enterprise.util.AnnotationLiteral;
+import javax.sql.DataSource;
+
+import org.quartz.utils.PoolingConnectionProvider;
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.arc.InstanceHandle;
+
+public class QuarkusQuartzConnectionPoolProvider implements PoolingConnectionProvider {
+    private AgroalDataSource dataSource;
+    private static String dataSourceName;
+
+    public QuarkusQuartzConnectionPoolProvider() {
+        InstanceHandle<AgroalDataSource> instanceHandle;
+        ArcContainer container = Arc.container();
+        boolean useDefaultDataSource = "QUARKUS_QUARTZ_DEFAULT_DATASOURCE".equals(dataSourceName);
+        if (useDefaultDataSource) {
+            instanceHandle = container.instance(AgroalDataSource.class);
+        } else {
+            instanceHandle = container.instance(AgroalDataSource.class, new DataSourceLiteral(dataSourceName));
+        }
+        if (instanceHandle.isAvailable()) {
+            this.dataSource = instanceHandle.get();
+        } else {
+            String message = String.format(
+                    "JDBC Store configured but '%s' datasource is missing. You can configure your datasource by following the guide available at: https://quarkus.io/guides/datasource-guide",
+                    useDefaultDataSource ? "default" : dataSourceName);
+            throw new IllegalStateException(message);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public QuarkusQuartzConnectionPoolProvider(Properties properties) {
```

Comment:
What is this supposed to do?

Reply:
Ah following our chat in Zulip, this class went through quite a lot of changes, it was previously a `Substitute` and this ctor was a substitute of the default connection provider from Quartz. After several iterations, this can indeed be removed. Thank you for catching this one.